sql: fix ALTER PRIMARY KEY with virtual columns on the table

otan opened the pull request:
Release note (bug fix): Fix a bug where ALTER PRIMARY KEY would fail if
the table contained virtual columns.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/62843)
<!-- Reviewable:end -->

rafiss reviewed (approved):
yay testing!! nice find

btw, optimizer team just added VIRTUAL columns this release, so i think only the 21.1 backport is needed.

otan commented on pkg/sql/logictest/testdata/logic_test/alter_primary_key:
```diff
@@ -1404,3 +1404,24 @@ drop table child3;
+                         CONSTRAINT "primary" PRIMARY KEY (new_pk ASC),
+                         UNIQUE INDEX table_with_virtual_cols_id_key (id ASC),
+                         FAMILY fam_0_new_pk (new_pk),
+                         FAMILY fam_1_id (id)
```
hehe, this randomised family stuff man

otan committed:
sql: fix ALTER PRIMARY KEY with virtual columns on the table

Release note (bug fix): Fix a bug where ALTER PRIMARY KEY would fail if
the table contained virtual columns.

otan commented:
bors r=rafiss

otan commented:
bors r=rafiss

RaduBerinde commented:
Nice find, thanks for fixing!
